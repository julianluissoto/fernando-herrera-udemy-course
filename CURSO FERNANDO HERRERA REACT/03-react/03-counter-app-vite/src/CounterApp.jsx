import React from 'react'
import { useState } from 'react'
import PropTypes from "prop-types"

/* function handleAdd (event) {
    console.log(event)
    } */
const CounterApp = ({value}) => {

 const handleAdd = ()=>{
        setCounter(counter+1)
    }
    
 const handleSustract = ()=>{
    setCounter(counter-1)
 }
 const handleReset = ()=>{
    setCounter(value)
 }

const [counter, setCounter]= useState(value)
  return (
    <div >
        <h1> Counter App </h1>
        <h2>  { counter } </h2>
        <button onClick={handleAdd
         }> +1 </button>
         <button onClick={handleSustract}> -1 </button>
         <button aria-label='btn-reset'  onClick={handleReset}> Reset </button>
    </div>
  )
}
CounterApp.propTypes={
value:PropTypes.number

}

export default CounterApp