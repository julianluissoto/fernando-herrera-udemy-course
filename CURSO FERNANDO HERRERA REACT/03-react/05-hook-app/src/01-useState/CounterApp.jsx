import { useState } from "react";

const CounterApp = () => {
  const [state, setState] = useState({
    Counter1: 10,
    Counter2: 20,
    Counter3: 30,
  });
  const { Counter1, Counter2, Counter3 } = state;
  const onAddCounter = () => {
    setState({
      ...state,
      Counter1: state.Counter1 + 1,
    });
  };

  return (
    <>
      <h1>counter app</h1>
      <hr />
      <h2>Counter: {Counter1}</h2>
      <h2>Counter: {Counter2}</h2>
      <h2>Counter: {Counter3}</h2>
      <hr />

      <button className="btn btn-success" onClick={onAddCounter}>
        {" "}
        set counter 1 en +1
      </button>
    </>
  );
};

export default CounterApp;
