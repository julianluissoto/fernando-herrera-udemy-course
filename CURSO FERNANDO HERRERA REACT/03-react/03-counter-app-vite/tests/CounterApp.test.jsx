import CounterApp from "../src/CounterApp";
import { fireEvent, render, screen } from '@testing-library/react'


describe('TEST ON <CounterApp/>', () => { 

    const initialValue= 100

    test('should match with the snapshot ', () => {
    
        const {container} = render(<CounterApp value={initialValue} />)
          expect(container).toMatchSnapshot();
    });
    test('should match should show inital value 100 ', () => {
    
       render(<CounterApp value={initialValue}/>)
       expect(screen.getByText(100)).toBeTruthy()
         // otra forma seria asi, buscando en el nodo html
         //  expect(screen.getByRole('heading', {level:2}).innerHTML).toContain(initialValue.toString());
    });
    test('shoul increment the counter on 1  ', () => {

        render(<CounterApp value={initialValue}/>)
        fireEvent.click( screen.getByText('+1') )
expect(screen.getByText(101)).toBeTruthy()
    });
    test('shoul decrement the counter on 1  ', () => {

        render(<CounterApp value={initialValue}/>)
        fireEvent.click( screen.getByText('-1') )
expect(screen.getByText(99)).toBeTruthy()
    });
    test('shoul reset the counter to the initial value  ', () => {

        render(<CounterApp value={50}/>)
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByText('+1') )
        fireEvent.click( screen.getByRole("button",{name:'btn-reset'})) 
       expect(screen.getByText('50')).toBeTruthy()


    });

 })