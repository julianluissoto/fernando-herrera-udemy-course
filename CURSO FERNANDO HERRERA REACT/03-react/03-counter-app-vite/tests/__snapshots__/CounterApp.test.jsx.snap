// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEST ON <CounterApp/> should match with the snapshot  1`] = `
<div>
  <div>
    <h1>
       Counter App 
    </h1>
    <h2>
        
      100
       
    </h2>
    <button>
       +1 
    </button>
    <button>
       -1 
    </button>
    <button
      aria-label="btn-reset"
    >
       Reset 
    </button>
  </div>
</div>
`;
